### ISSUE TITLE
KtLint 0.40.0 has changed mechanism for file path loading, we face errors when .editorconfig is missing

### ISSUE THREAD

**Issue by petertrr** [diktat-static-analysis/diKTat#616]
We can't merge #615 now, because file_path is not present in user data. Ktlint now uses editorconfig (`EditorConfigLoader.loadPropertiesForFile`), we need to workaround it somehow.

**Comment by petertrr** [comment]
Similar bugs are reported in ktlint already: https://github.com/pinterest/ktlint/issues/997

**Comment by petertrr** [comment]
They have solved this on their side, once ktlint 0.41 is out is should be safe for us to upgrade.

**Comment by petertrr** [comment]
For ktlint 0.41, different configuration is required for gradle dependency: https://github.com/pinterest/ktlint/issues/1114

**Comment by dimsuz** [comment]
But this configuration also results in the following warning:

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.jetbrains.kotlin.com.intellij.util.ReflectionUtil (file:/home/***/.gradle/caches/modules-2/files-2.1/org.jetbrains.kotlin/kotlin-compiler-embeddable/1.4.31/6451ea797cef544e81f507b0d9959cd97ae09c0/kotlin-compiler-embeddable-1.4.31.jar) to method java.util.ResourceBundle.setParent(java.util.ResourceBundle)
WARNING: Please consider reporting this to the maintainers of org.jetbrains.kotlin.com.intellij.util.ReflectionUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

**Comment by akuleshov7** [comment]
@petertrr what do you think? Are these warnings related to ktlint?

**Comment by akuleshov7** [comment]
It looks like it is a bug in Kotlin compiler: https://youtrack.jetbrains.com/issue/KT-43520

**Comment by petertrr** [comment]
@dimsuz  this warning indeed originates from ktlint, or even from kotlin compiler. It has been there in previous versions too, and it is indeed a warning, i.e. ktlint still works as expected, until this will be deprecated in JVM.

**Comment by dimsuz** [comment]
I see, I'll star it, thanks!
